Gcp/service/firestore

liu1234636 opened the pull request:
Sorry for I had been pushing from a forked repo instead of a cloned branch. These are all from my previous PR but now are coming from a clone - all previous feedback has been addressed. Documentation is completed.

liu1234636 committed:
git main

liu1234636 committed:
git main and policy

liu1234636 committed:
git main and policy

liu1234636 committed:
git main change

liu1234636 committed:
git main change

liu1234636 committed:
git main change

liu1234636 committed:
Copy fork resp structure into new branch and raise a new pr

liu1234636 committed:
chore: remove all .DS_Store files

liu1234636 committed:
chore: ignore .DS_Store

liu1234636 committed:
Merge branch 'dev' into gcp/service/Firestore

# Conflicts:
#	inputs/gcp/cloud_vmware_engine/external_access_rule/block_broad_external_access/plan
#	inputs/gcp/cloud_vmware_engine/external_access_rule/block_broad_external_access/plan.json
#	inputs/gcp/cloud_vmware_engine/network/allowed_location/plan
#	inputs/gcp/cloud_vmware_engine/network/allowed_location/plan.json
#	inputs/gcp/cloud_vmware_engine/network/block_legacy/plan
#	inputs/gcp/cloud_vmware_engine/network/block_legacy/plan.json
#	inputs/gcp/cloud_vmware_engine/network_peering/block_custom_routes/plan
#	inputs/gcp/cloud_vmware_engine/network_peering/block_custom_routes/plan.json
#	inputs/gcp/cloud_vmware_engine/network_peering/block_thirdparty/plan
#	inputs/gcp/cloud_vmware_engine/network_peering/block_thirdparty/plan.json
#	inputs/gcp/cloud_vmware_engine/network_policy/allowed_location/plan
#	inputs/gcp/cloud_vmware_engine/network_policy/allowed_location/plan.json
#	inputs/gcp/cloud_vmware_engine/network_policy/block_external_ip/plan
#	inputs/gcp/cloud_vmware_engine/network_policy/block_external_ip/plan.json
#	inputs/gcp/cloud_vmware_engine/network_policy/block_internet_access/plan
#	inputs/gcp/cloud_vmware_engine/network_policy/block_internet_access/plan.json
#	inputs/gcp/cloud_vmware_engine/private_cloud/allowed_location/plan
#	inputs/gcp/cloud_vmware_engine/private_cloud/allowed_location/plan.json
#	inputs/gcp/cloud_vmware_engine/private_cloud/zonal_location/plan
#	inputs/gcp/cloud_vmware_engine/private_cloud/zonal_location/plan.json
#	inputs/gcp/parameter_manager/parameter/encryption/plan
#	inputs/gcp/parameter_manager/parameter/encryption/plan.json
#	inputs/gcp/parameter_manager/regional_parameter/allowed_location/plan
#	inputs/gcp/parameter_manager/regional_parameter/allowed_location/plan.json
#	inputs/gcp/parameter_manager/regional_parameter/encryption/plan
#	inputs/gcp/parameter_manager/regional_parameter/encryption/plan.json
#	inputs/gcp/scc/event_threat_detection_custom_module/enablement_state/.terraform.lock.hcl
#	inputs/gcp/scc/event_threat_detection_custom_module/enablement_state/plan
#	inputs/gcp/scc/event_threat_detection_custom_module/enablement_state/plan.json
#	inputs/gcp/scc/folder_custom_module/enablement_state/.terraform.lock.hcl
#	inputs/gcp/scc/folder_custom_module/enablement_state/plan
#	inputs/gcp/scc/folder_custom_module/enablement_state/plan.json
#	inputs/gcp/scc/mute_config/type/.terraform.lock.hcl
#	inputs/gcp/scc/mute_config/type/plan
#	inputs/gcp/scc/mute_config/type/plan.json
#	inputs/gcp/scc/organization_custom_module/enablement_state/.terraform.lock.hcl
#	inputs/gcp/scc/organization_custom_module/enablement_state/plan
#	inputs/gcp/scc/organization_custom_module/enablement_state/plan.json
#	inputs/gcp/scc/project_custom_module/enablement_state/.terraform.lock.hcl
#	inputs/gcp/scc/project_custom_module/enablement_state/plan
#	inputs/gcp/scc/project_custom_module/enablement_state/plan.json

liu1234636 committed:
Fix some need

liu1234636 committed:
fix conflict

liu1234636 committed:
Merge branch 'dev' into gcp/service/Firestore

# Conflicts:
#	inputs/gcp/google_kms/google_kms_crypto_key/scheduled_destroy_duration/config.tf
#	inputs/gcp/google_kms/google_kms_crypto_key/scheduled_rotation_period/config.tf
#	inputs/gcp/scc/mute_config/type/config.tf

Seb-sketch commented:
Please ensure that you are not deleting any existing files in your PR, we cannot merge one that will remove the services from other students or any main files that are not part of your service.

Seb-sketch reviewed (changes requested):
Please ensure that you are not deleting any existing files in your PR, we cannot merge one that will remove the services from other students or any main files that are not part of your service.

liu1234636 committed:
fix some bugs

liu1234636 committed:
fix [Terraform failed to compile!] bugs
fix [Unmentioned resources other than 'c' found: nc1] bugs

liu1234636 committed:
fix [Terraform failed to compile!] bugs
fix [Resources in output other than 'nc' found: c] bugs

liu1234636 committed:
fix [Terraform failed to compile!] bugs

liu1234636 committed:
fix [Terraform failed to compile!] bugs

liu1234636 committed:
fix [Terraform failed to compile!] bugs

liu1234636 committed:
fix [Service: firestore | Resource: firestore_backup_schedule | Policy: daily_recurrence
Unmentioned resources other than 'c' found: nc] bugs

liu1234636 committed:
fix [Terraform failed to compile!] bugs

liu1234636 committed:
Merge remote-tracking branch 'origin/dev' into gcp/service/Firestore

# Conflicts:
#	.gitignore
#	policies/gcp/cloud_vmware_engine/external_access_rule/block_broad_external_access/policy.rego
#	policies/gcp/cloud_vmware_engine/external_access_rule/vars.rego
#	policies/gcp/cloud_vmware_engine/network/allowed_location/policy.rego
#	policies/gcp/cloud_vmware_engine/network/block_legacy/policy.rego
#	policies/gcp/cloud_vmware_engine/network/vars.rego
#	policies/gcp/cloud_vmware_engine/network_peering/block_custom_routes/policy.rego
#	policies/gcp/cloud_vmware_engine/network_peering/block_thirdparty/policy.rego
#	policies/gcp/cloud_vmware_engine/network_peering/vars.rego
#	policies/gcp/cloud_vmware_engine/network_policy/allowed_location/policy.rego
#	policies/gcp/cloud_vmware_engine/network_policy/block_external_ip/policy.rego
#	policies/gcp/cloud_vmware_engine/network_policy/block_internet_access/policy.rego
#	policies/gcp/cloud_vmware_engine/network_policy/vars.rego
#	policies/gcp/cloud_vmware_engine/private_cloud/allowed_location/policy.rego
#	policies/gcp/cloud_vmware_engine/private_cloud/vars.rego
#	policies/gcp/cloud_vmware_engine/private_cloud/zonal_location/policy.rego
#	policies/gcp/parameter_manager/parameter/encryption/policy.rego
#	policies/gcp/parameter_manager/parameter/vars.rego
#	policies/gcp/parameter_manager/regional_parameter/allowed_location/policy.rego
#	policies/gcp/parameter_manager/regional_parameter/encryption/policy.rego
#	policies/gcp/parameter_manager/regional_parameter/vars.rego
#	policies/gcp/scc/event_threat_detection_custom_module/enablement_state/policy.rego
#	policies/gcp/scc/event_threat_detection_custom_module/vars.rego
#	policies/gcp/scc/folder_custom_module/enablement_state/policy.rego
#	policies/gcp/scc/folder_custom_module/vars.rego
#	policies/gcp/scc/organization_custom_module/enablement_state/policy.rego
#	policies/gcp/scc/organization_custom_module/vars.rego
#	policies/gcp/scc/project_custom_module/enablement_state/policy.rego
#	policies/gcp/scc/project_custom_module/vars.rego

liu1234636 committed:
fix [auto_text] bugs

liu1234636 committed:
fix [_helpers: can't find format_range_validation_message] bugs

Seb-sketch commented on policies/gcp/firestore/firestore_backup_schedule/daily_recurrence/policy.rego:
You have a comment here with a correct type of policy but your set policy type is not one of the five listed within the helpers.rego.

liu1234636 replied:
Thanks. Modified.

Seb-sketch commented on policies/gcp/firestore/firestore_backup_schedule/retention/policy.rego:
min_duration is not one of the policy types defined in the helpers

liu1234636 replied:
Thanks. Modified.

Seb-sketch commented on policies/gcp/firestore/firestore_backup_schedule/weekly_recurrence/policy.rego:
existence is not one of the policy types defined in the helpers

liu1234636 replied:
Thanks. Modified.

Seb-sketch commented on policies/gcp/firestore/firestore_document/fields/policy.rego:
contains_all is not one of the policy types defined in the helpers

liu1234636 replied:
Thanks. Modified.

Seb-sketch reviewed (changes requested):
Your names for many c and nc files are dc or dnc, this is causing autotest issues and need to be changed, also ensure to get rid of unnecessary comments within files both c/nc and policy files. 

Ensure the policy types you're using are 1 of the 5 defined within the helpers.

We don't want PR's to have deletion of other exisiting policies, this PR is deleting many DS_Store files and inputs/policies for cloud_vmware_engine and parameter_manager we don't want this happening.

liu1234636 committed:
fix [policy_type] bugs

liu1234636 committed:
remove dc in firestore_backup_schedule

liu1234636 committed:
fix some bug

liu1234636 committed:
change c.tf and nc.tf
add vars.rego

liu1234636 committed:
1.fix bug:
Service: firestore | Resource: firestore_backup_schedule | Policy: weekly_recurrence
Resources in output other than 'nc' found: c

2.fix bug:
Service: firestore | Resource: firestore_backup_schedule | Policy: daily_recurrence
Unmentioned resources other than 'c' found: nc

3.fix bug:
Service: firestore | Resource: firestore_document | Policy: collection/fields/project
Could not run OPA query!

liu1234636 committed:
1.fix bug:
Service: firestore | Resource: firestore_backup_schedule | Policy: retention
Resources in output other than 'nc' found: c

2.fix bug:
Service: firestore | Resource: firestore_document | Policy: fields
Resources in output other than 'nc' found: c

liu1234636 committed:
1.fix bug:
Service: firestore | Resource: firestore_backup_schedule | Policy: retention
Resources in output other than 'nc' found: c

2.fix bug:
Service: firestore | Resource: firestore_document | Policy: fields
Resources in output other than 'nc' found: c

3. fix bug:
Service: firestore | Resource: firestore_backup_schedule | Policy: weekly_recurrence
Terraform failed to compile!

Service: firestore | Resource: firestore_backup_schedule | Policy: daily_recurrence
Terraform failed to compile!

liu1234636 committed:
fix bug:
Service: firestore | Resource: firestore_document | Policy: fields
Resources in output other than 'nc' found: c

liu1234636 committed:
fix bugs:
Service: firestore | Resource: firestore_backup_schedule | Policy: daily_recurrence
Terraform failed to compile!

liu1234636 committed:
fix bugs:
Service: firestore | Resource: firestore_backup_schedule | Policy: weekly_recurrence
Terraform failed to compile!

Service: firestore | Resource: firestore_backup_schedule | Policy: retention
Terraform failed to compile!

liu1234636 committed:
fix bugs:

liu1234636 committed:
fix bugs:
Service: firestore | Resource: firestore_backup_schedule | Policy: weekly_recurrence
Resources in output other than 'nc' found: c

liu1234636 committed:
fix bugs:
Service: firestore | Resource: firestore_backup_schedule | Policy: weekly_recurrence
Resources in output other than 'nc' found: c

liu1234636 committed:
fix bugs:
Service: firestore | Resource: firestore_backup_schedule | Policy: daily_recurrence
Unmentioned resources other than 'c' found: nc

Service: google_bigquery_datapolicy_data_policy | Resource: data_masking_policy | Policy: predefined_expression
Unmentioned resources other than 'c' found: nc

liu1234636 committed:
fix bugs:
Service: firestore | Resource: firestore_backup_schedule | Policy: weekly_recurrence
Could not run OPA query!

liu1234636 committed:
revert

liu1234636 committed:
fix bugs:
Service: firestore | Resource: firestore_backup_schedule | Policy: weekly_recurrence
Resources in output other than 'nc' found: c

liu1234636 committed:
1.fix bugs:
Service: firestore | Resource: firestore_backup_schedule | Policy: daily_recurrence
Unmentioned resources other than 'c' found: nc

2.delete useless code

liu1234636 committed:
fix bugs:
Service: firestore | Resource: firestore_backup_schedule | Policy: retention
Unmentioned resources other than 'c' found: nc

liu1234636 committed:
1. fix bugs:
Service: firestore | Resource: firestore_document | Policy: fields
Resources in output other than 'nc' found: c

2.fix bugs:
message = null

liu1234636 committed:
merge dev

liu1234636 commented:
Hello teacher, all the policy.rego in my service: firestore have passed the test in the cloud. But now still error in other student's service and my merging is blocked.

liu1234636 commented:
![Uploading policy_firestore.png…]()

Seb-sketch commented:
@liu1234636 That should be fine if other's are having issues, as long as your firestore policies are passing the checks and it seems they are. Please ensure that in your PR there are no deleting of files, the .DS files, inputs and policies are being removed and if merged they will be deleted, we do not want this to happen.

liu1234636 commented:
> @liu1234636 That should be fine if other's are having issues, as long as your firestore policies are passing the checks and it seems they are. Please ensure that in your PR there are no deleting of files, the .DS files, inputs and policies are being removed and if merged they will be deleted, we do not want this to happen.

Thanks. I made sure I didn't delete files, i just merge all the dev in my branch. I removed all .DS files & inputs when uploading my  code so there will not have .DS & inputs in my code.

Seb-sketch commented:
> > @liu1234636 That should be fine if other's are having issues, as long as your firestore policies are passing the checks and it seems they are. Please ensure that in your PR there are no deleting of files, the .DS files, inputs and policies are being removed and if merged they will be deleted, we do not want this to happen.
> 
> Thanks. I made sure I didn't delete files, i just merge all the dev in my branch. I removed all .DS files & inputs when uploading my code so there will not have .DS & inputs in my code.

For these changes we don't want the removal of any config files for scc or parameter_manager or any other services.we want the configs back. 
A merge won't work to restore them either as the change is downstream it will need to be a git reset or manually moving the files back in place. Other than this small change everything seems good to go though

liu1234636 commented:
> > > @liu1234636 That should be fine if other's are having issues, as long as your firestore policies are passing the checks and it seems they are. Please ensure that in your PR there are no deleting of files, the .DS files, inputs and policies are being removed and if merged they will be deleted, we do not want this to happen.
> > 
> > 
> > Thanks. I made sure I didn't delete files, i just merge all the dev in my branch. I removed all .DS files & inputs when uploading my code so there will not have .DS & inputs in my code.
> 
> For these changes we don't want the removal of any config files for scc or parameter_manager or any other services.we want the configs back. A merge won't work to restore them either as the change is downstream it will need to be a git reset or manually moving the files back in place. Other than this small change everything seems good to go though

@Seb-sketch 
Thanks, I will restore the file use command:
git diff --name-status origin/dev | ForEach-Object {
    if ($_ -match "^D\s+(.+)$") {
        git checkout origin/dev -- $matches[1]
    }
}

liu1234636 committed:
Merge branch 'dev' into gcp/service/Firestore

liu1234636 committed:
restore files,use command:
git diff --name-status origin/dev | ForEach-Object {
    if ($_ -match "^D\s+(.+)$") {
        git checkout origin/dev -- $matches[1]
    }
}

liu1234636 commented:
@Seb-sketch All the file have been restored.

liu1234636 committed:
fix a bug

liu1234636 committed:
Delete useless comments

liu1234636 committed:
delete not c/nc in .tf files

Seb-sketch commented on inputs/gcp/firestore/firestore_backup_schedule/daily_recurrence/c.tf:
I do not know why you have made so many of these lines into comments, all you needed to do was to make name = "c"

liu1234636 replied:
Thanks, changed

Seb-sketch commented on inputs/gcp/firestore/firestore_backup_schedule/daily_recurrence/nc.tf:
project or name should equal "nc"

liu1234636 replied:
Thanks,changed.

Seb-sketch commented on inputs/gcp/firestore/firestore_backup_schedule/retention/nc.tf:
should have set name = "nc"

liu1234636 replied:
Thanks,changed.

Seb-sketch reviewed (changes requested):
You did not need to make the changes done here, there was no need to make so much of them as comments, all you needed to do was to make the name equal to c or nc respectively. You have done this correctly in your files for firestore_database, all you have to do is the same in firestore_backup_schedule and firestore_document. set name = "c" or "nc" depending on which file

liu1234636 committed:
Change name or project = c/nc
Change policy

liu1234636 commented:
@Seb-sketch 
Hi teacher, I have change all the name/project to c or nc. My  OPA commands are getting the nc as non-compliant resources.

My policy can pass in my local test, but it still failed in CI env. [Terraform failed to compile!] I check autp_test.py, it may cause by terraform init or terraform plan -out=tfplan failed. But in local env they work.

My test screenshots are as folled, 9 policies:
<img width="1080" height="689" alt="1" src="https://github.com/user-attachments/assets/ffb17e05-e314-4342-8efc-fbc9cc833b0c" />

<img width="1080" height="689" alt="2" src="https://github.com/user-attachments/assets/67873dc0-f2da-458b-b8bd-2d6795adb672" />

<img width="1080" height="689" alt="3" src="https://github.com/user-attachments/assets/64de954f-4c07-4734-a110-2ed7323b9512" />

<img width="1080" height="689" alt="4" src="https://github.com/user-attachments/assets/8f4c1131-8b0d-420a-90ed-ef737a81a9fd" />

<img width="1080" height="689" alt="5" src="https://github.com/user-attachments/assets/36329e52-a581-4785-b29d-6a9de981fd80" />

<img width="1080" height="689" alt="6" src="https://github.com/user-attachments/assets/0770929b-5b04-444e-ba5a-f823c1606fda" />

<img width="1080" height="689" alt="7" src="https://github.com/user-attachments/assets/d961a2cd-0b2f-450a-825b-efd9b60ccbc9" />

<img width="1080" height="689" alt="8" src="https://github.com/user-attachments/assets/47be7ae4-63d8-4be1-a938-2c5fff0bdf1c" />

<img width="1080" height="689" alt="9" src="https://github.com/user-attachments/assets/06d1ed4d-bc05-4a9f-ae13-4e73f6370cef" />

liu1234636 commented:
@Seb-sketch 
Sorry for  i upload [1.fields & 8.fields] test pic. It should be [8.fields]. 
[1.daily_re]pic is as followed:
<img width="1080" height="689" alt="1" src="https://github.com/user-attachments/assets/3c834b28-b049-4502-b477-9f00bca2892e" />

Seb-sketch commented:
<img width="1088" height="688" alt="Screenshot 2025-09-26 131730" src="https://github.com/user-attachments/assets/437c8f57-400a-4cfc-afb3-a7a4f73ecea5" />

Seb-sketch reviewed (approved):
Thank you for addressing the changes, following proper naming syntax and making sure file removals aren't being pushed any more. It's unfortunate the autotest isn't working for your policies but the screenshots show them catching the nc files successfully. Thank you for also providing the daily_recurrence screenshot in teams.
